js/siteorigin-panels/view/row.js

```diff
@@ -161,14 +161,47 @@ module.exports = Backbone.View.extend( {
 
 		var duplicateRow = this.model.clone( this.builder.model );
 
-		this.builder.model.rows.add( duplicateRow, {
-			at: this.builder.model.rows.indexOf( this.model ) + 1
+		this.builder.model.get('rows').add( duplicateRow, {
+			at: this.builder.model.get('rows').indexOf( this.model ) + 1
 		} );
 
 		this.builder.model.refreshPanelsData();
 	},
 
 	/**
+	 * Copy the row to a localStorage
+	 */
+	copyHandler: function(){
+		if ( typeof(Storage) === "undefined" || ! panelsOptions.user ) return;
+
+		var serial = panels.serial.serialize( this.model );
+		serial.thingType = 'row-model';
+
+		// Store this in the cookie
+		localStorage[ 'panels_clipboard_' + panelsOptions.user ] = JSON.stringify( serial );
+	},
+
+	/**
+	 * Create a new row and insert it
+	 */
+	pasteHandler: function(){
+		if ( typeof(Storage) === "undefined" || ! panelsOptions.user ) return;
```

Comment:
Yeah, agree there. It would also make it much easier to change the backend of the clipboard (for example, to an actual clipboard).